src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(n) {
+    if (n === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+    sum += n;
+
+    return adder;
```

Comment:
Consider adding a check to ensure that the input `n` is a number before adding it to `sum`. This will prevent potential errors if non-numeric values are passed to the `adder` function.